samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Services/AgentServices.cs

```diff
@@ -0,0 +1,433 @@
+            
+            // Return the generated images as JSON
+            return System.Text.Json.JsonSerializer.Serialize(generatedImages);
+        }
+        catch (Exception ex)
+        {
+            Logger.LogError(ex, "Error processing image descriptions");
+            return "[]";
+        }
+    }
+    
+    /// <summary>
+    /// Adds a placeholder image to the list of generated images
+    /// </summary>
+    private void AddPlaceholderImage(List<GeneratedImage> images, ImageDescription description)
+    {
+        images.Add(new GeneratedImage
+        {
+            Description = description.Description,
+            Prompt = description.Prompt,
+            ImageUrl = "https://via.placeholder.com/800x600?text=Image+Generation+Sample",
+            Caption = description.Caption
+        });
+        
+        Logger.LogInformation($"Added placeholder image for: {description.Description}");
+    }
+    
+    /// <summary>
+    /// Calls the DALL-E API to generate an image from a prompt using Microsoft Entra ID authentication
+    /// </summary>
+    /// <param name="prompt">The image generation prompt</param>
+    /// <returns>URL of the generated image</returns>
+    private async Task<string> GenerateImageWithDallE(string prompt)
+    {
+        if (!_dalleEnabled || _httpClient == null || _dallEEndpoint == null || _credential == null)
+        {
+            throw new InvalidOperationException("DALL-E is not enabled. Check environment variable DALLE_ENDPOINT.");
+        }
+        
+        // Create the request body for DALL-E API
+        var requestBody = new
+        {
+            prompt = prompt,
+            n = 1,
+            size = "1024x1024",
+            response_format = "url",
+            quality = "standard"
+        };
+        
+        var content = new StringContent(
+            System.Text.Json.JsonSerializer.Serialize(requestBody), 
+            System.Text.Encoding.UTF8, 
+            "application/json");
+            
+        Logger.LogInformation($"Using DALL-E endpoint directly: {_dallEEndpoint}");
+        Logger.LogInformation($"Calling DALL-E API with prompt: {prompt}");
+        
+        // Get an access token from Azure using DefaultAzureCredential
+        var tokenRequestContext = new Azure.Core.TokenRequestContext(
+            scopes: new[] { "https://cognitiveservices.azure.com/.default" });
+        var accessToken = await _credential.GetTokenAsync(tokenRequestContext);
+        
+        // Add the bearer token to the request header
+        _httpClient.DefaultRequestHeaders.Authorization = 
+            new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", accessToken.Token);
+        
+        // Make the API call - use the full endpoint URL directly from the environment variable
+        // The DALLE_ENDPOINT should already contain the complete URL including deployment name and API version
+        var response = await _httpClient.PostAsync(_dallEEndpoint, content);
```

Comment:
`Response` implements the `IDisposable` interface and therefore need to be disposed when you're done using them.
```suggestion
        using var response = await _httpClient.PostAsync(_dallEEndpoint, content);
```